distribution/pull_v2.go

```diff
@@ -392,9 +392,13 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named, platform
 		if p.config.RequireSchema2 {
 			return false, fmt.Errorf("invalid manifest: not schema2")
 		}
-		msg := schema1DeprecationMessage(ref)
-		logrus.Warn(msg)
-		progress.Message(p.config.ProgressOutput, "", msg)
+
+		// only warn on Docker Hub, to incite upgrading
+		if reference.Domain(ref) == "docker.io" {
```

Comment:
I'm not sure I understand this.
Why would someone update an old image and why should someone pester other people to do so?

Reply:
And even more so, why would it be specific to hub?

Reply:
This is specific to the hub because we know the timeframe of which the hub supported schema2. These represent old images which have not received any updates since that point. After push is disabled for all registries, then we can reconsider the deprecation notice for all schema 1 manifests. Even if they are still supported, schema1 manifests will be much slower in the future and suggesting to update the image is a valid suggestion. While the original message was incorrect, it did show that there are hub users who may be unknowingly using very outdated images. It may be worthwhile in the future to display such a message on any outdated images, for example older official images, if it is possible to make such a determination.

Reply:
That's why I think it would be worth putting behind a header; other registries (such as Quay) might want to add this header at some point if we deem it worth dropping support for Schema 2 V1. I suspect that support drop will be FAR into the future, but it would be nice to have the option.

Reply:
I'm afraid the header would need updates to distribution too. A comment was added to explain why Docker Hub is hard-coded. Also reworded the message for clarity.